Crashlytics/UnitTests/FIRCLSSettingsTests.m

```diff
@@ -419,4 +419,59 @@ - (void)testCorruptCacheKey {
   XCTAssertEqual(self.settings.errorLogBufferSize, 64 * 1000);
 }
 
+- (void)testNewReportEndpointSettings {
+  NSString *settingsJSON =
+      @"{\"settings_version\":3,\"cache_duration\":60,\"app\":{\"status\":\"activated\",\"update_"
+      @"required\":false,\"report_upload_variant\":2}}";
+
+  NSError *error = nil;
+  [self writeSettings:settingsJSON error:&error];
+  NSTimeInterval currentTimestamp = [NSDate timeIntervalSinceReferenceDate];
+  [self.settings cacheSettingsWithGoogleAppID:TestGoogleAppID currentTimestamp:currentTimestamp];
+
+  XCTAssertNil(error, "%@", error);
+  XCTAssertTrue(self.settings.shouldUseNewReportEndpoint);
```

Comment:
Can you add this assertion (with the right default) to `testDefaultSettings` and `testCorruptCache`?

Reply:
Done